doc/Iothub_sdk_options.md

```diff
@@ -0,0 +1,77 @@
+```
+
+An example of using a set option:
+
+```c
+bool trace_on = true;
+IoTHubClient_SetOption(sdk_handle, OPTION_LOG_TRACE, &trace_on);
+
+HTTP_PROXY_OPTIONS http_proxy;
+memset(&http_proxy, 0, sizeof(HTTP_PROXY_OPTIONS));
+http_proxy.host_address = PROXY_ADDRESS;
+http_proxy.port = PROXY_PORT;
+DPS_LL_SetOption(handle, OPTION_HTTP_PROXY, &http_proxy);
+```
+
+## Available Option
+
+<a name="IotHub_option"></a>
+
+### IoThub_Client
+
+| Option Name        | Option Define              | Value Type         | Description
+|--------------------|----------------------------|--------------------|-------------------------------
+| `"messageTimeout"` | OPTION_MESSAGE_TIMEOUT     | tickcounter_ms_t*  | Timeout used for message
+| `"product_info"`   | OPTION_PRODUCT_INFO        | const char*        | Product information sent to the IoThub service on connection
+| `"TrustedCerts"`   | OPTION_TRUSTED_CERT        | const char*        | Azure Server certificate used to validate TLS connection to iothub
+
+<a name="transport_option"></a>
+
+### Standard Transport
+
+| Option Name            | Option Define             | Value Type         | Description
+|------------------------|---------------------------|--------------------|-------------------------------
+| `"logtrace"`           | OPTION_LOG_TRACE          | bool* value        | Turn on and off log tracing for the transport
+| `"sas_token_lifetime"` | OPTION_SAS_TOKEN_LIFETIME | `size_t`* value    | Length of time in seconds used for lifetime of sas token.
+| `"x509certificate"`    | OPTION_X509_CERT          | const char*        | Sets the x509 certificate used for connection authentication
+| `"x509privatekey"`     | OPTION_X509_PRIVATE_KEY   | const char*        | Sets the private key for the x509 certificate
+| `"x509EccCertificate"` | OPTION_X509_ECC_CERT      | const char*        | Sets the ECC x509 certificate used for connection authentication
+| `"x509EccAliasKey"`    | OPTION_X509_ECC_KEY       | const char*        | Sets the private key for the ECC x509 certificate
+| `"proxy_data"`         | OPTION_HTTP_PROXY         | [HTTP_PROXY_OPTIONS*][http-proxy-object]| Http proxy data object used for proxy connection to IoTHub
+
+### MQTT Transport
+
+| Option Name            | Option Define             | Value Type         | Description
+|------------------------|---------------------------|--------------------|-------------------------------
+| `"keepalive"`          | OPTION_KEEP_ALIVE         | int*               | Length of time to send `Keep Alives` to service for D2C Messages
+
+### AMQP Transport
+
+| Option Name                  | Option Define                   | Value Type        | Description
+|------------------------------|---------------------------------|-------------------|-------------------------------
+| `"cbs_request_timeout"`      | OPTION_CBS_REQUEST_TIMEOUT      | `size_t`* value   | Length of time for CBS token to expire
```

Comment:
>CBS token to expire [](start = 106, length = 19)

I don't think this is for the token expiration. I believe this is for the put request over CBS. Please confirm.